### PR TITLE
Support the legacy boot GPT flag

### PR DESCRIPTION
**Pull request by intrigeri**
This is needed to fix a regression in Udisks
(https://github.com/storaged-project/udisks/issues/415).

Fixes: #287

### CODE REVIEW

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

File: src/lib/plugin_apis/part.api [hunk 1]
```diff
@@ -49,7 +49,8 @@ typedef enum {
     BD_PART_FLAG_GPT_SYSTEM_PART = 1 << 25,
     BD_PART_FLAG_GPT_READ_ONLY = 1 << 26,
     BD_PART_FLAG_GPT_HIDDEN = 1 << 27,
-    BD_PART_FLAG_GPT_NO_AUTOMOUNT = 1 << 28
+    BD_PART_FLAG_GPT_NO_AUTOMOUNT = 1 << 28,
+    BD_PART_FLAG_GPT_LEGACY_BOOT = 1 << 29
```
Comment (vpodzime) [inline src/lib/plugin_apis/part.api]: The flag is already defined above as ``BD_PART_FLAG_LEGACY_BOOT``. It's one of the flags supported by *libparted*.

**Comment by intrigeri** [comment]
> Please use the existing definition of the flag.
> The flag is already defined above as ``BD_PART_FLAG_LEGACY_BOOT``. It's one of the flags supported by *libparted*.

Thanks for the hint, done locally. Before I update this PR, do you
prefer an additional commit on top of my original branch, or a v2
branch that squashes the new change with my original one? The latter
results in a cleaner history, while the former might save you some
review time. Your call.

> Thanks for taking a look at this!

Thanks for being so welcoming :)

**Comment by vpodzime** [comment]
> The latter results in a cleaner history, while the former might save you some review time. Your call.

I prefer cleaner history over some extra review time. Please squash the changes together and force-push the result to your branch to update this PR. Thanks!

**Comment by vpodzime** [comment]
@intrigeri, do you have any estimate of when you'll be able to apply the suggested changes? I'd love to make this part of the next release which is scheduled for Tuesday October 31st.

**Comment by intrigeri** [comment]
> @intrigeri, do you have any estimate of when you'll be able to apply the suggested changes?
> I'd love to make this part of the next release which is scheduled for Tuesday October 31st.

I had booked time to work on this again on Nov 5-8. But given this release schedule I could make time for it on Oct 29. Would this work for you or is there some kind of freeze that starts earlier?

**Comment by vpodzime** [comment]
> I had booked time to work on this again on Nov 5-8. But given this release schedule I could make time for it on Oct 29. Would this work for you or is there some kind of freeze that starts earlier?

There's no freeze. Oct 29/30 would be great! Thanks!

**Comment by intrigeri** [comment]
Done, force-pushed!

**Pull request merged**